File: src/main/java/com/vc/hard/L972.scala
package com.vc.hard

object L972 {
  def isRationalEqual(S: String, T: String): Boolean = {
    solve(S) == solve(T)
  }

  def solve(T: String): Double = {
    val toi = T.indexOf("(")
    if(toi != -1) {
      var tBase = T.substring(0, toi)
      val tDecimal = T.substring(toi + 1, T.length - 1)
      (0 to 20).foreach(i => tBase += tDecimal)
      return tBase.toDouble
    }
    T.toDouble
  }
}

File: src/main/java/com/vc/hard/L972.java
package com.vc.hard;

class L972 {
    public boolean isRationalEqual(String S, String T) {
        return solve(S) == solve(T);
    }

    private double solve(String s) {
        int index = s.indexOf("(");
        if(index != -1) {
            String tBase = s.substring(0, index);
            String tDecimal = s.substring(index + 1, s.length() - 1);
            s = tBase;
            for(int i = 0; i < 20; i++) s += tDecimal;
        }
        return Double.parseDouble(s);
    }
}
